doc/SAI-Proposal-ACL-1.md

```diff
@@ -0,0 +1,89 @@
+SAI access control lists (ACL) enhancements for SAI 1.0 release
+-------------------------------------------------------------------------------
+
+ Title       | SAI ACL Model - Enhancements
+-------------|----------------------
+ Authors     | Cavium Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 08/09/2016
+ SAI-Version | 1.0.0
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+SAI Access Control Lists (ACL) object implements ACL management functions. In SAI 0.9.1 through 0.9.5 versions, SAI ACL contained three types of objects, ACL table, ACL entry and ACL counter. The ACL table contains a number of ACL entries. Each ACL table defines a set of unique matching fields for all its ACL entries. A packet can match rules in different ACL tables and take non-conflicting actions from all the matched rules. However, within an ACL table, if a packet matches multiple rules, only the actions from the rule of highest priority are executed. ACL counters can also be created and attached to an ACL entry in order to counter the number of packets or bytes that match the ACL entry. The initial version of ACL table object has several ambiguities and limitations. We propose the following enhancements to address some of those and propose a generic and simple ACL model for operators:
```

Comment:
@tushar-ty : I will update the spec with the examples that you have asked for. Please review and let me know if you have further queries. 

The benefits of the new ACL model for SAI 1.0.0, and the motivation to enhance/update the current ACL model (pre SAI 1.0) is very clearly mentioned in the beginning of the section: Again listing for reference in this comment:

This proposal introduced for SAI 1.0.0, proposes the following enhancements:       
1. Introduce well defined binding point for an ACL table (or ACL Group) to be applied
2. Clarify the behavior and usage of ACL group ID and metadata fields              
3. Simplify ACL table stages that were relevant in absense of a binding point   
4. Address scaling concerns in absense of a well defined binding point             
5. Introduce behavioral model for ACLs                                             
6. Adding tunnel and bridge interface specific ACL behavior